src/Microsoft.ML.StandardLearners/Standard/LogisticRegression/LbfgsPredictorBase.cs

```diff
@@ -164,18 +184,24 @@ internal LbfgsTrainerBase(IHostEnvironment env, TArgs args, SchemaShape.Column l
             Contracts.CheckUserArg(Args.SgdInitializationTolerance >= 0, nameof(Args.SgdInitializationTolerance), "Must be non-negative");
             Contracts.CheckUserArg(Args.NumThreads == null || Args.NumThreads.Value >= 0, nameof(Args.NumThreads), "Must be non-negative");
 
-            L2Weight = Args.L2Weight;
-            L1Weight = Args.L1Weight;
-            OptTol = Args.OptTol;
-            MemorySize = Args.MemorySize;
+            Contracts.CheckParam(l2Weight >= 0, nameof(l2Weight), "Must be non-negative, if provided.");
```

Comment:
>Contracts [](start = 12, length = 9)

why not Host?

Reply:
thanks!

---
In reply to: [221114697](https://github.com/dotnet/machinelearning/pull/1084#discussion_r221114697) [](ancestors = 221114697)